presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixInitializer.java

```diff
@@ -84,14 +88,14 @@ public void initializeRubix(NodeManager nodeManager)
                         rubixConfigurationInitializer.setMasterAddress(master.getHostAndPort());
                         rubixConfigurationInitializer.setCurrentNodeAddress(nodeManager.getCurrentNode().getHost());
 
-                        Configuration configuration = new Configuration(false);
-                        for (ConfigurationInitializer configurationInitializer : configurationInitializers) {
-                            configurationInitializer.initializeConfiguration(configuration);
-                        }
-
-                        // RubixConfigurationInitializer.initializeConfiguration will not update configurations yet as it has not been fully initialized
-                        // Apply configurations from it by skipping init check
+                        Configuration configuration = getInitialConfiguration();
+                        // Perform standard hdfs configuration initialization.
+                        // This will also call out to RubixConfigurationInitializer but this will be no-op because
```

Comment:
dependency cycle... do you see a way to avoid it?

Reply:
Not sure if that is possible really (at least I do not see any simple way to do that). The dependency on other `ConfigurationInitializers` we have here is not just of technical nature. We cannot just drop it. We really need to run all those initializers before we can start-up Rubix infrastructure. Theoretically we could try to create `Provider<RubixConfigurationInitializer>` and do initialization there, but I do not believe it would fly:
a) how can we inject other `ConfigurationInitializers` to provider?
b) Rubix initialization require node to be part of the cluster. I would expect guice bootstrapping happens before that....